backend/tests/integration-async/disambiguation/test_disambiguation_tasks.py

```diff
@@ -765,9 +772,100 @@ def assert_authors_records_exist_in_es():
+                {
+                    "full_name": "Kowal, Michal",
+                    "ids": [{"schema": "INSPIRE BAI", "value": "J.M.Maldacena.1"}],
+                    "affiliations": [{"value": "Warsaw U."}],
+                    "record": {"$ref": "http://localhost:5000/api/authors/999101"},
+                    "curated_relation": True,
+                }
+            ]
+        }
+    )
+    literature_record = LiteratureRecord.create(data=literature_data)
+
+    literature_data_2 = faker.record("lit", with_control_number=True)
+    literature_data_2.update(
+        {
+            "authors": [
+                {
+                    "full_name": "Kowal, Michal",
+                    "ids": [{"schema": "INSPIRE BAI", "value": "J.M.Maldacena.2"}],
+                    "record": {"$ref": "http://localhost:5000/api/authors/999102"},
+                    "curated_relation": True,
+                }
+            ]
+        }
+    )
+    literature_record_2 = LiteratureRecord.create(data=literature_data_2)
+
+    db.session.commit()
+
+    def assert_authors_records_exist_in_es():
+        lit_record_from_es = InspireSearch.get_record_data_from_es(literature_record)
+        lit_record_from_es_2 = InspireSearch.get_record_data_from_es(
+            literature_record_2
+        )
+        assert lit_record_from_es
+        assert lit_record_from_es_2
+
+    retry_until_pass(assert_authors_records_exist_in_es)
+
+    literature_data_3 = faker.record("lit", with_control_number=True)
+    literature_data_3.update({"authors": [{"full_name": "Kowal, Michal"}]})
+    literature_record_3 = LiteratureRecord.create(data=literature_data_3)
+    db.session.commit()
+
+    def assert_first_disambiguation_no_match():
+        literature_record_from_es = InspireSearch.get_record_data_from_es(
+            literature_record_3
+        )
+
+        assert get_values_for_schema(
+            literature_record_from_es["authors"][0]["ids"], "INSPIRE BAI"
+        )
+        assert (
+            literature_record_from_es["authors"][0]["ids"]
+            != literature_record["authors"][0]["ids"]
+        )
+        assert (
+            literature_record_from_es["authors"][0]["ids"]
+            != literature_record_2["authors"][0]["ids"]
+        )
+
+    retry_until_pass(assert_first_disambiguation_no_match, retry_interval=2)
+
+    db.session.expire_all()
+    lit_record = InspireRecord.get_record(literature_record_3.id)
+    lit_record["authors"][0]["affiliations"] = [{"value": "Warsaw U."}]
+    lit_record.update(dict(lit_record))
+    db.session.commit()
+
+    def assert_disambiguation_on_record_update():
+        literature_record_from_es = InspireSearch.get_record_data_from_es(
+            literature_record_3
+        )
+        assert (
+            literature_record_from_es["authors"][0]["ids"]
+            == literature_record["authors"][0]["ids"]
+        )
+
+    retry_until_pass(assert_disambiguation_on_record_update, retry_interval=2)
```

Comment:
I think there is one more test case to consider: if an author record is created on record creation due to an ambiguous match, and the match is still ambiguous during update, it remains assigned to the same record instead of creating yet another one.